### PR TITLE
On window blur, immediately hide keyboard hints

### LINKED ISSUE #14526: [ui] Keyboard nav: disable hints on window defocus
If you're trying to cmd+shift+4 on a mac for a screenshot, for example, keyboard hints persist. Disable them when you're not focused on the browser/window in question.

### PR DESCRIPTION
**Pull request by philrenaud**
Resolves #14526

Solves an issue where you can have shift held and click away and hints are still shown.

This will **not** fix this for when you're taking screenshots, as cmd+shift+5 (and etc.) doesn't fire any window-or-document-detectable events, but it's still an improvement.

### CODE REVIEW

File: ui/app/templates/csi/volumes/index.hbs [hunk 1]
```diff
@@ -38,7 +38,7 @@
         @source={{p.list}}
         @sortProperty={{this.sortProperty}}
         @sortDescending={{this.sortDescending}}
-        @class="with-foot {{if this.keyboard.displayHints "display-hints"}}" as |t|
+        @class="with-foot" as |t|
```
Comment (philrenaud) [inline ui/app/templates/csi/volumes/index.hbs:41]: Side-effect: .display-hints was not being used for anything anymore.

File: ui/app/components/keyboard-shortcuts-modal.js [hunk 2]
```diff
@@ -8,6 +9,13 @@ export default class KeyboardShortcutsModalComponent extends Component {
   @service keyboard;
   @service config;
 
+  constructor() {
+    super(...arguments);
+    window.addEventListener('blur', () => {
+      set(this, 'keyboard.displayHints', false);
```
Comment (ChaiWithJai) [inline ui/app/components/keyboard-shortcuts-modal.js:15]: `question`: Should this get cleaned up after the component unmounts?

**Review by ChaiWithJai — APPROVED** [review]
I think this technically has a memory leak because we don't remove the event listener when the component unmounts. This might probably be better off as an element modifier. But its no big deal and probably not worth the effort.

**Comment by philrenaud** [comment]
> I think this technically has a memory leak

@ChaiWithJai How might you fix it, if it were worth doing? This is an app-level always-on component.

File: ui/app/components/keyboard-shortcuts-modal.js (hunk 2 above)
Comment (philrenaud) [reply ui/app/components/keyboard-shortcuts-modal.js:15]: I don't think so. displayHints = false is the default / at-rest state.

**Pull request merged**